libethereum/BlockChain.cpp

```diff
@@ -1375,18 +1379,26 @@ bool BlockChain::isKnown(h256 const& _hash, bool _isCurrent) const
 	DEV_READ_GUARDED(x_blocks)
 		if (!m_blocks.count(_hash))
 		{
-			string d;
-			m_blocksDB->Get(m_readOptions, toSlice(_hash), &d);
-			if (d.empty())
+			try
+			{
+				m_blocksDB->lookup(toSlice(_hash));
```

Comment:
Better to replace this with a call to `exists`

Reply:
Overall, `exists` followed by `lookup` effectively does 2 lookups, whereas `lookup` that might throw will only do one lookup.

Reply:
But I agree in this case seeing as the value is discarded.